### ISSUE TITLE
Device client not working on UWP using AMQP

### ISSUE THREAD

**Issue by Heris19** [Azure/azure-iot-sdk-csharp#421]
- **OS, version, SKU and CPU architecture used:** <VERSION> Windows 10 Desktop x64
- **Application's .NET Target Framework :** <NETVERSION> uap10.0, 10.0.16299.0
- **Device:** <DEVICETYPE> Development machine
- **SDK version used:** <VERSION> Master branch, commit dcc9be4c Adding dev guide and retry policy documentation. / Microsoft.Azure.Devices.Client v1.7.0

# Description of the issue:
DeviceClient is not working when used inside UWP using AMQP, it generates a ton of following exceptions:
```
Exception thrown: 'System.MissingMethodException' in Microsoft.Azure.Devices.Client.dll
Exception thrown: 'System.MissingMethodException' in System.Private.CoreLib.dll
Exception thrown: 'System.MissingMethodException' in System.Private.CoreLib.dll
Exception thrown: 'System.MissingMethodException' in System.Private.CoreLib.dll
```
Until it throws this exception:
```
An exception of type 'Microsoft.Azure.Devices.Client.Exceptions.IotHubCommunicationException' occurred in Microsoft.Azure.Devices.Client.dll but was not handled in user code
Timed out trying to create FaultTolerantAmqpObject`1
```
When I cloned the repository and linked libraries directly instead of nuget packages, I was able to catch this exception:
```
Method 'TlsTransportSettings.set_Certificate(X509Certificate2)' from assembly 'Microsoft.Azure.Amqp' was not included in compilation, but was referenced in IotHubConnection.CreateTlsTransportSettings(). 
There may have been a missing assembly, or a dependency on a more recent Windows SDK release.
```
With stack trace:
```
   at Microsoft.Azure.Devices.Client.IotHubConnection.CreateTlsTransportSettings()
   at Microsoft.Azure.Devices.Client.IotHubConnection.<CreateSessionAsync>d__30.MoveNext() in C:\Users\herudek\Source\Repos\azure-iot-sdk-csharp\iothub\device\src\IotHubConnection.cs:line 269
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() in f:\dd\ndp\fxcore\CoreRT\src\System.Private.CoreLib\src\System\Runtime\ExceptionServices\ExceptionDispatchInfo.cs:line 61
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) in f:\dd\ndp\fxcore\CoreRT\src\System.Private.CoreLib\src\System\Runtime\CompilerServices\TaskAwaiter.cs:line 178
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) in f:\dd\ndp\fxcore\CoreRT\src\System.Private.CoreLib\src\System\Runtime\CompilerServices\TaskAwaiter.cs:line 147
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task) in f:\dd\ndp\fxcore\CoreRT\src\System.Private.CoreLib\src\System\Runtime\CompilerServices\TaskAwaiter.cs:line 119
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult() in f:\dd\ndp\fxcore\CoreRT\src\System.Private.CoreLib\src\System\Runtime\CompilerServices\TaskAwaiter.cs:line 476
   at Microsoft.Azure.Devices.Client.IotHubSingleTokenConnection.<CreateSessionAsync>d__9.MoveNext() in C:\Users\herudek\Source\Repos\azure-iot-sdk-csharp\iothub\device\src\IotHubSingleTokenConnection.cs:line 55
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() in f:\dd\ndp\fxcore\CoreRT\src\System.Private.CoreLib\src\System\Runtime\ExceptionServices\ExceptionDispatchInfo.cs:line 61
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) in f:\dd\ndp\fxcore\CoreRT\src\System.Private.CoreLib\src\System\Runtime\CompilerServices\TaskAwaiter.cs:line 178
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) in f:\dd\ndp\fxcore\CoreRT\src\System.Private.CoreLib\src\System\Runtime\CompilerServices\TaskAwaiter.cs:line 147
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task) in f:\dd\ndp\fxcore\CoreRT\src\System.Private.CoreLib\src\System\Runtime\CompilerServices\TaskAwaiter.cs:line 119
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult() in f:\dd\ndp\fxcore\CoreRT\src\System.Private.CoreLib\src\System\Runtime\CompilerServices\TaskAwaiter.cs:line 476
   at Microsoft.Azure.Devices.Client.FaultTolerantAmqpObject`1.<OnCreateAsync>d__5.MoveNext() in C:\Users\herudek\Source\Repos\azure-iot-sdk-csharp\iothub\device\src\Common\FaultTolerantAmqpObject.cs:line 45
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw() in f:\dd\ndp\fxcore\CoreRT\src\System.Private.CoreLib\src\System\Runtime\ExceptionServices\ExceptionDispatchInfo.cs:line 61
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) in f:\dd\ndp\fxcore\CoreRT\src\System.Private.CoreLib\src\System\Runtime\CompilerServices\TaskAwaiter.cs:line 178
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) in f:\dd\ndp\fxcore\CoreRT\src\System.Private.CoreLib\src\System\Runtime\CompilerServices\TaskAwaiter.cs:line 147
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task) in f:\dd\ndp\fxcore\CoreRT\src\System.Private.CoreLib\src\System\Runtime\CompilerServices\TaskAwaiter.cs:line 119
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult() in f:\dd\ndp\fxcore\CoreRT\src\System.Private.CoreLib\src\System\Runtime\CompilerServices\TaskAwaiter.cs:line 476
   at Microsoft.Azure.Devices.Client.Singleton`1.<CreateValue>d__19.MoveNext() in C:\Users\herudek\Source\Repos\azure-iot-sdk-csharp\iothub\device\src\Common\Singleton.cs:line 160
```

There are conditional symbols for Certificate property in [https://github.com/Azure/azure-amqp](https://github.com/Azure/azure-amqp) TlsTransportSetting ommiting Certificate for UWP.

Basically we have been dealing with issues on UWP for a some time, especially on ARM, but with the newest packages we are having these problems.

# Code sample exhibiting the issue:
Add to newly created UWP project
```
var authentication = AuthenticationMethodFactory.CreateAuthenticationWithRegistrySymmetricKey("ID", "Key");
var deviceClient = DeviceClient.Create("HubAddress", authentication, TransportType.Amqp);

var message = new Message(Encoding.ASCII.GetBytes(""));
await deviceClient.SendEventAsync(message).ConfigureAwait(false);
```

**Comment by CIPop** [comment]
Thanks for reporting @Heris19 . I've added this to our Xamarin stabilization project.

**Comment by Heris19** [comment]
Hello @CIPop, @abhipsaMisra 

it's been a few months since I reported this issue, do you have any progress? 

Thank you

**Comment by emnu1971** [comment]
Hi all,

check my post on stackoverflow for possible resolution:

https://stackoverflow.com/questions/53681825/exception-thrown-system-missingmethodexception-in-microsoft-azure-devices-cli

Kr,
Emmanuel Nuyttens

**Comment by prmathur-microsoft** [comment]
We will be tackling Xamarin related started January. At the moment we are happy to review your PR if this issue is blocking you.

**Comment by tiwaripradeep123** [comment]
any update on your stmt "We will be tackling Xamarin related started January. " - Do you have new package out with changes or some timelines for this ?

**Comment by jenoola** [comment]
@tiwaripradeep123 could you please retry with the lastest SDK version and check if you are still observing MissingMethodException? Or is it a different one now?

**Comment by prmathur-microsoft** [comment]
Can you please update us if you see this issue with latest 1.19.0 SDK ? If not we would like to close this.

**Comment by Heris19** [comment]
The issue is still there, I have even tried target version 10.0.17763.0, but nothing has changed.

**Comment by prmathur-microsoft** [comment]
We have root cause this to an issue in underlying library amqp and this is where we are tracking the issue https://github.com/Azure/azure-amqp/issues/120

**Comment by jenoola** [comment]
@Heris19 Please try to add Microsoft.Azure.Amqp 2.4.1 to your project, it should fix the issue

**Comment by jenoola** [comment]
Fix checked in the new release 1.20.0

**Comment by az-iot-builder-01** [comment]
@CIPop, @Heris19, @emnu1971, @prmathur-microsoft, @tiwaripradeep123, @jenoola, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
